feeds/src/contracts/FluxAggregatorContract.js

```diff
@@ -0,0 +1,310 @@
+import { ethers } from 'ethers'
+import {
+  getLogs,
+  formatAnswer,
+  decodeLog,
+  createContract,
+  createInfuraProvider,
+} from './utils'
+import _ from 'lodash'
+
+export default class PrepaidContract {
+  answerUpdatedEvent = {
+    filter: {},
+    listener: {},
+  }
+  submissionReceivedEvent = {
+    filter: {},
+    listener: {},
+  }
+
+  newRoundEvent = {
+    filter: {},
+    listener: {},
+  }
+
+  answerIdInterval
+  provider
+  contract
+
+  constructor(options, abi) {
+    this.provider = createInfuraProvider(options.networkId)
+    this.contract = createContract(options.contractAddress, this.provider, abi)
+    this.address = options.contractAddress
+    this.alive = true
+    this.abi = abi
+    this.options = options
+  }
+
+  kill() {
+    try {
+      if (!this.alive) return false
+      clearInterval(this.answerIdInterval)
+      this.removeListener(
+        this.answerUpdatedEvent.filter,
+        this.answerUpdatedEvent.listener,
+      )
+      this.contract = null
+      this.address = null
+      this.alive = false
+      this.options = null
+    } catch (error) {
+      //
+    }
+  }
+
+  removeListener(filter, eventListener) {
+    if (!this.alive) return
+
+    this.provider.removeListener(filter, eventListener)
+  }
+
+  async oracles() {
+    return await this.contract.getOracles()
+  }
+
+  async minimumAnswers() {
+    const minAnswers = await this.contract.minAnswerCount()
+    return Number(minAnswers)
+  }
+
+  async latestRound() {
+    const latestRound = await this.contract.latestRound()
+    return Number(latestRound)
+  }
+
+  async reportingRound() {
+    const reportingRound = await this.contract.reportingRound()
+    return Number(reportingRound)
+  }
+
+  async latestAnswer() {
+    const latestAnswer = await this.contract.latestAnswer()
+    return formatAnswer(
+      latestAnswer,
+      this.options.multiply,
+      this.options.decimalPlaces,
+    )
+  }
+
+  async latestTimestamp() {
+    const latestTimestamp = await this.contract.latestTimestamp()
+    return Number(latestTimestamp)
+  }
+
+  async getAnswer(answerId) {
+    const getAnswer = await this.contract.getAnswer(answerId)
+    return formatAnswer(
+      getAnswer,
+      this.options.multiply,
+      this.options.decimalPlaces,
+    )
+  }
+
+  async getTimestamp(answerId) {
+    const timestamp = await this.contract.getTimestamp(answerId)
+    return Number(timestamp)
+  }
+
+  async description() {
+    const description = await this.contract.description()
+    return ethers.utils.parseBytes32String(description)
+  }
+
+  async decimals() {
+    const decimals = await this.contract.decimals()
+    return Number(ethers.utils.bigNumberify(decimals))
```

Comment:
Should we use parseInt here?
```suggestion
    return ethers.utils.bigNumberify(decimals).toNumber()
```